jenkins-plugin/src/main/java/org/jenkinsci/plugins/pipeline/maven/WithMavenStepExecution.java

```diff
@@ -654,11 +659,22 @@ private String setupSettingFile(@Nonnull Collection<Credentials> credentials) th
             return settingsDest.getRemote();
         }
 
-        // Settings provided by the global maven configuration
-        SettingsProvider settingsProvider = GlobalMavenConfig.get().getSettingsProvider();
+        SettingsProvider settingsProvider;
+
+        MavenConfigFolderOverrideProperty overrideProperty = getMavenConfigOverrideProperty();
+        if (overrideProperty != null) {
+            // Settings overriden by a folder property
+            console.format("[withMaven] using overriden Maven settings by folder '%s'%n", overrideProperty.getOwner().getDisplayName());
+            settingsProvider = overrideProperty.getSettings();
+        } else {
+            console.format("[withMaven] using Maven settings provided by the Jenkins global configuration%n");

```

Comment:
Can we output this message only if Level.FINE is enabled?
See  [JENKINS-48657 - Noisy build log message "\[withMaven\] Skip 'xxx' disabled by configuration"](https://issues.jenkins-ci.org/browse/JENKINS-48657)